sklearn/manifold/t_sne.py

```diff
@@ -674,6 +678,14 @@ def __init__(self, n_components=2, perplexity=30.0,
         self.method = method
         self.angle = angle
 
+        # TODO sklearn 0.20: remove this warning
+        if learning_rate is None:
+            import warnings
+            warnings.warn("The default learning rate of TSNE has changed " +
+                          "from 1000 to 200 in sklearn 0.19. Set the " +
+                          "learning rate explicitely to avoid this message.")
+            self.learning_rate = 200.0
```

Comment:
The only problem is that you're doing this in `__init__` when you should be doing it in `fit`.